internal/content_coding.go

```diff
@@ -188,6 +195,21 @@ func (e *ZlibEncoder) Encode(data []byte) ([]byte, error) {
 	return e.buf.Bytes(), nil
 }
 
+type ZstdEncoder struct {
+	encoder *zstd.Encoder
+}
+
+func NewZstdEncoder(level int) (*ZstdEncoder, error) {
+	e, err := zstd.NewWriter(nil, zstd.WithEncoderLevel(zstd.EncoderLevelFromZstd(level)))
+	return &ZstdEncoder{
+		encoder: e,
+	}, err
+}
+
+func (e *ZstdEncoder) Encode(data []byte) ([]byte, error) {
+	return e.encoder.EncodeAll(data, make([]byte, 0, len(data))), nil
+}
+
```

Comment:
Could we split this into yet another PR. Sorry for being so fine-grained, but mixing topics in the PR usually leads to trouble when trying to pin-down errors introduced by some commits.